src/main/webapp/app/overview/course-overview.component.ts

```diff
@@ -276,6 +301,32 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
         }
     }
 
+    /** initialize courses attribute by retreiving recently accessed courses from the server */
+    getRecentlyAccessedCourses() {
+        this.courseService.findAllForDashboard().subscribe({
+            next: (res: HttpResponse<CoursesForDashboardDTO>) => {
+                if (res.body) {
+                    const courses: Course[] = [];
+                    for (const courseDto of res.body.courses) {
+                        courses.push(courseDto.course);
+                    }
```

Comment:
I think that this could be simplified

```suggestion
                    const { courses: courseDtos } = res.body;
                    const courses = courseDtos.map(courseDto => courseDto.course);
```

Reply:
agreed